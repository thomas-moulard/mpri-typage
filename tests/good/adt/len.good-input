type list = 
  Nil
| Cons of int * list
end

val ( + ) (x : int) (y : int) = 0

val rec len l = 
  match l with
    | Nil -> 0 
    | Cons x xs -> 1 + len xs
  end  

val l = len Nil
val l' = len (Cons 0 Nil)
