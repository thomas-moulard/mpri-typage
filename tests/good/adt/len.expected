Nil = Nil

Cons = Cons

+ = fun (x : int).fun (y : int).0

len = fun l.match l with 
    | Nil -> 0
    | Cons x xs -> 1 (fun (x1 : int).(fun (y : int).0)) (let rec len1 l1 =
      match l1 with 
          | Nil -> 0
          | Cons x1 xs1 -> ((1 (fun (x2 : int).(fun (y : int).0))) (len1 xs1))
      end

    in
len1 xs)
end


l = 0

l' = 0

l' : int
l : int
len : (list -> int)
+ : (int -> (int -> int))
Cons : (int -> (list -> list))
Nil : list
